grid/address/geocoding/maps/here.suma

```diff
@@ -0,0 +1,144 @@
+profile = "address/geocoding@3.1"
+provider = "here"
+
+// Here API documentation
+//   https://developer.here.com/documentation
+
+map Geocode {
+    payload = input.query || `${input.streetAddress || ""} ${input.addressLocality || ""} ${input.addressRegion || ""} ${input.postalCode || ""} ${input.addressCountry || ""}`;
+
+    http GET "geocode" "/v1/geocode" {
+        security "apiKey"
+        
+        request {
+            query {
+                q = payload
+            }
+        }
+
+        response 200 "application/json" {
+            return map error if (body.items.length === 0) {
+                title = 'Results not found',
+                detail = 'No results were found for specified address'
+            }
+            
+            map result {
+                latitude = body.items[0].position.lat
+                longitude = body.items[0].position.lng
+            }
+        }
+
+        response 400 {
+            map error {
+                title = body.title
+                detail = body.cause
+            }
+        }
+
+        response 405 {
```

Comment:
Is not this mistake, i did. not find 405 status code in HERE doc: https://developer.here.com/documentation/geocoder/dev_guide/topics/http-status-codes.html

Other status codes handling is missing here (401,...)

Reply:
Their documentation is confusing, I'm working with their v7 API, even though I specify v1 in URL 🤷🏻‍♂️ [Here](https://developer.here.com/documentation/geocoding-search-api/api-reference-swagger.html) they have just 400, 405, 429 and 503 codes.

Reply:
It's probably based on their services and I'm using `HERE Geocoding & Search API`

Reply:
I got 401 when tried without valid api key:
<img width="762" alt="image" src="https://user-images.githubusercontent.com/5206165/154476399-b37f4456-1874-4990-beac-593172ed48f7.png">

Reply:
> Their documentation is confusing, I'm working with their v7 API, even though I specify v1 in URL 🤷🏻‍♂️ [Here](https://developer.here.com/documentation/geocoding-search-api/api-reference-swagger.html) they have just 400, 405, 429 and 503 codes.

But I was looking into wrong docs, you are right.

Reply:
Even the body content of the error returned by HERE API is different from the doc you linked above.

<img width="408" alt="image" src="https://user-images.githubusercontent.com/5206165/154477269-8ba3c783-53cd-4695-8319-0435c794292e.png">

Reply:
But my link to status codes is wrong. It points to XML API.

Reply:
That's weird, I'll add 401 and 403 just in case.